cli/src/cast.rs

```diff
@@ -232,17 +232,30 @@ async fn main() -> eyre::Result<()> {
             legacy,
             confirmations,
             to_json,
+            resend,
         } => {
             let provider = Provider::try_from(eth.rpc_url()?)?;
             let chain_id = Cast::new(&provider).chain_id().await?;
             let sig = sig.unwrap_or_default();
 
             if let Ok(Some(signer)) = eth.signer_with(chain_id, provider.clone()).await {
+                let from = match &signer {
+                    WalletType::Ledger(leger) => leger.address(),
+                    WalletType::Local(local) => local.address(),
+                    WalletType::Trezor(trezor) => trezor.address(),
+                };
+
+                let nonce = if resend {
+                    Some(provider.get_transaction_count(from, None).await?)
+                } else {
+                    nonce
+                };
```

Comment:
make `mut nonce` and then 
`nonce = Some(provider.get_transaction_count(from, None).await?);`